### ISSUE TITLE
BoxEncoder `scale` argument is a no-op?

### ISSUE THREAD

**Issue by matthew-frank** [NVIDIA/DALI#1819]
The `BoxEncoder` operation has an argument `scale` which is described in the docs as:
* scale (float, optional, default = 1.0) – Rescale the box and anchors values before offset calculation (e.g. to get back to absolute values).

But in the code it just seems to get multiplied into both the numerator and divisor of a fraction, so always cancels out.

In box_encoder.cc the scales are multiplied into box[i] and anchor[i] at line 127, and then they cancel out in divides in the return statement at line 130.
In box_encoder.cu it's similar in the `MatchOffsets` function at line 84.

**Comment by JanuszL** [comment]
@awolant could you take a look?

**Comment by matthew-frank** [comment]
Just to be clear: I don't understand what the intended use case is for the `scale` argument.  The only place it is currently used at all is if `offset=True`.  But `offset=True` already (a) normalizes everything to the height and width of the corresponding anchor (i.e., there is no absolute scale to scale to), and (b) the `offset=True` option already takes the `manes` and `stds` arguments for offset "normalization", so there's already a multiplicative factor (that is not a no-op) applied to the output when `offset=True`.  (`stds` doesn't really "normalize" the offsets, rather it _scales_ them (see https://github.com/mlperf/training/blob/master/single_stage_detector/ssd/base_model.py#L147 and L161).  It's actually not clear to me why the mplerf reference uses this scaling (the scaling from what Dali calls `stds`).  Perhaps to deal with underflow issues, or maybe it's intended to help scale the relative importance of the location loss to the classification loss.)

**Comment by awolant** [comment]
@matthew-frank is right. I'm not sure why this parameter was introduced here. Maybe @Kh4L remebres? Wasn't it done to match Gluon implementation or something like that? I worked only on PyTorch one and I don't think it was there.

**Comment by JanuszL** [comment]
Indeed it was added in https://github.com/NVIDIA/DALI/pull/921.

**Comment by Kh4L** [comment]
IIRC, I did it to be more numerically close to the paper, back when I was adding DALI to gluon-cv model zoo’s SSD.
